pkg/manager/internal.go

```diff
@@ -389,17 +365,18 @@ func (cm *controllerManager) serveMetrics(stop <-chan struct{}) {
 		Handler: mux,
 	}
 	// Run the server
-	go func() {
+	cm.startRunnable(RunnableFunc(func(stop <-chan struct{}) error {
 		log.Info("starting metrics server", "path", defaultMetricsEndpoint)
 		if err := server.Serve(cm.metricsListener); err != nil && err != http.ErrServerClosed {
-			cm.errSignal.SignalError(err)
+			return err
 		}
-	}()
+		return nil
+	}))
 
 	// Shutdown the server when stop is closed
 	<-stop
 	if err := server.Shutdown(context.Background()); err != nil {
```

Comment:
Do we still want to use `context.Background` here or give it a context with a timeout?

Reply:
So the reason I left this with `context.Background` is that we use channels directly and not a `context.Context` to handle the graceful shutdown.

I guess it would be possible to use a `context.WithTimeout` and propagate it but then we have to select on its `Done()` and afterwards return an error only if its `Error() == context.DeadlineExceeded()`. WDYT?

Reply:
sgtm, probably worth a comment for posterity

Reply:
:+1: on contexts, but can maybe do that overhaul in a follow-up?  Whatever you feel better about